### PR TITLE
KeySerializer used to serialize attribute map keys in Resource

### PR DESCRIPTION
**Pull request by duseja2**

### CODE REVIEW

**Comment by DeathByTape** [comment]
https://github.com/yahoo/elide/issues/198

**Comment by yahoocla** [comment]
CLA is valid!

File: elide-core/src/main/java/com/yahoo/elide/jsonapi/serialization/KeySerializer.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            return;
+        } else if (cls == Class.class) {
+            str = ((Class<?>) value).getName();
+        } else if (cls.isEnum()) {
+            str = ((Enum<?>) value).name();
+        }
+        else {
+            str = value.toString();
+        }
+        jgen.writeFieldName(str);    }
```
Comment (clayreimann) [inline elide-core/src/main/java/com/yahoo/elide/jsonapi/serialization/KeySerializer.java]: Closing brace should be on newline.

File: elide-core/src/main/java/com/yahoo/elide/jsonapi/serialization/KeySerializer.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdKeySerializer;
+
+import java.io.IOException;
+import java.util.Date;
+
+/**
+ * Custom serializer for Serializing Map Keys.
```
Comment (clayreimann) [inline elide-core/src/main/java/com/yahoo/elide/jsonapi/serialization/KeySerializer.java]: Could you provide more clarification for why this is here/what it does?

**Comment by DeathByTape** [comment]
:+1:

**Pull request merged**